command/registry.go

```diff
@@ -193,8 +193,12 @@ func createConfiguredBucket(templateDir string, opts ...bucketConfigurationOpts)
 
 	if bucket.Slug == "" {
 		diags = append(diags, &hcl.Diagnostic{
-			Summary:  "bucket name cannot be empty",
-			Detail:   "empty bucket name, please set it with the HCP_PACKER_BUCKET_NAME environment variable, or in a `hcp_packer_registry` block",
+			Summary: "Image bucket name required",
+			Detail: "You must provide an image bucket name for HCP Packer builds. " +
+				"You can either set the HCP_PACKER_BUCKET_NAME environment " +
+				"variable or set the bucket_name argument in an hcp_packer_registry " +
+				"block (HCL2 templates only). If none of those are defined, and a " +
+				"build name is, it will become the bucket name.",
```

Comment:
Not sure if it’s the best call out but this also only applies to HCL2 templates
```suggestion
				"build name (HCL2 templates only) is, it will become the bucket name.",
```

Reply:
True, this is indeed HCL2-specific. I don't like the repetition however, maybe we could rephrase the whole HCL2 part?
Something like "On HCL2 templates, you can either set the bucket name in an `hcp_packer_registry` block, or if you provide a build name, it will become the bucket name."

Reply:
Yeah. I like that. I too am not a fan of the repetition. I would just make one change `or if you provide a name in the build block`

Reply:
Laura suggested another alternative, which does sound better, I've applied it in the latest reroll